{
  "cells": [
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "kQgPRQq3a4gN",
        "outputId": "10f043eb-47a0-4062-a645-939a11f2822d"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "\u001b[K     |████████████████████████████████| 9.2 MB 4.1 MB/s \n",
            "\u001b[K     |████████████████████████████████| 182 kB 45.5 MB/s \n",
            "\u001b[K     |████████████████████████████████| 237 kB 56.1 MB/s \n",
            "\u001b[K     |████████████████████████████████| 78 kB 5.9 MB/s \n",
            "\u001b[K     |████████████████████████████████| 4.7 MB 44.9 MB/s \n",
            "\u001b[K     |████████████████████████████████| 164 kB 48.6 MB/s \n",
            "\u001b[K     |████████████████████████████████| 63 kB 1.7 MB/s \n",
            "\u001b[K     |████████████████████████████████| 51 kB 6.3 MB/s \n",
            "\u001b[?25h  Building wheel for validators (setup.py) ... \u001b[?25l\u001b[?25hdone\n"
          ]
        }
      ],
      "source": [
        "!pip install streamlit -qqqq"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "kArwb9nUeZjw"
      },
      "outputs": [],
      "source": [
        "import streamlit as st\n",
        "st.title('AYUDANOS A CUIDAR EN MEDIO AMBIENTE')\n",
        "st.image(\"https://i.ytimg.com/vi/dGAMdXY3zX8/maxresdefault.jpg\", width=300)\n",
        "st.subheader(\"*$¿Qué$ $es$ $el$ $calentamiento$ $global?$*\")\n",
        "st.write(\"La Revolución Industrial fue el punto de inflexión en el que las emisiones de gases de efecto invernadero (GEI) arrojadas a la atmósfera empezaron a dispararse. Empezó a crecer la población y aumentó la demanda y producción de energía (obtenidas mayoritariamente a través de combustibles fósiles), lo que dio lugar a un nuevo modelo de producción y de consumo. El principal resultado fue el aumento global de la temperatura: de 1,1°C entre 1850 y 2017. En este articulo resumimos en qué situación nos encontramos en 2022.\")\n",
        "\n",
        "\n",
        "st.caption(\"-huella de carbono\")\n",
        "st.write(\"Mitigar los efectos del calentamiento global significa adaptar su estilo de vida a la situación actual, reduciendo y limitando las emisiones de gases de efecto invernadero. Para ello, el primer paso es tomar conciencia de su propia huella de carbono.\")\n",
        "st.write(\"El cálculo de la huella de carbono es el primer paso para reducirla: identificar las principales fuentes de emisiones de gases de efecto invernadero en nuestra vida cotidiana y adaptar así nuestro estilo de vida para minimizar su impacto de carbono en el medio ambiente; La reducción de las emisiones de gases de efecto invernadero es muy importante para luchar contra el calentamiento global: optimizar el consumo de energía, reducir nuestra huella digital de carbono y promover el transporte sostenible; La compensación de sus emisiones de carbono mediante la financiación de proyectos ambientales con el fin de reducir las emisiones de gases de efecto invernadero a la atmósfera y avanzar así hacia la neutralidad del carbono.\")\n",
        "\n",
        "st.sidebar.header(\"La contaminacion\")\n",
        "st.sidebar.subheader('El daño que hacen en el planeta')\n",
        "st.sidebar.write(\"Se entiende por contaminación ambiental cuando existe la presencia de sustancias nocivas en el agua, aire o suelo.\")\n",
        "st.sidebar.image(\"https://plusambiental.com/wp-content/uploads/2021/11/contaminacion-ambiental-todo.jpg\")\n",
        "\n",
        "st.sidebar.write(\"$tipos$ $de$ $contaminación$\")\n",
        "\n",
        "col1, col2 = st.columns([1,2])\n",
        "\n",
        "col1.header(\"Causas\")\n",
        "col2.header(\"Consecuencias\")\n",
        "\n",
        "st.header(\"$¿Qué$ $es$ $la$ $huella$ $ecologica?$\")\n",
        "st.write(\"El objetivo de reducir la huella ecológica es esencial debido al nivel de consumo de los recursos naturales y de la energía, la creación de desechos y las emisiones contaminantes por parte del ser humano. Se trata de tener claras nuestras necesidades reales y lo que requiere la naturaleza para su subsistencia. Hay que mantener presente que esta debe regenerarse y que, en ocasiones, tarda mucho tiempo en hacerlo.\")\n",
        "\n",
        "\n",
        "\n",
        "st.header(\"Reduce tu huella ecologica\")\n",
        "\n",
        "\n",
        "if st.sidebar.button(\"hidrica\"):\n",
        "  col1.write(\"Vertidos de aguas negras, Derrames de petróleo, Productos fitosanitarios, Deforestación, Aumento de temperatura\")\n",
        "  col2.write(\"Deterioro de la biodiversidad, Contaminación de la cadena alimentaria, Escasez de agua potable, Enfermedades, Mortalidad infantil\")\n",
        "  st.write(\"Ajustar el consumo de agua en casa\")\n",
        "  st.write(\"Las buenas prácticas en el hogar nunca están demás. Las duchas cortas, cerrar los caños cuando no se está usando, reducir el nivel de agua en las lavadoras, arreglar las llaves y tuberías de los lavaderos para que no goteen y regar las plantas cuando llueve son acciones que pueden parecer insignificantes, pero que en el tiempo ayudan en gran medida a reducir la huella hídrica y contribuir al cuidado del medio ambiente.\")\n",
        "  st.write(\"Reduce el consumo de carne\")\n",
        "  st.write(\"Aunque no lo parezca, cada kilogramo de carne bovina produce un gasto de más de 15 mil litros de agua, o sea unos 40 camiones cisterna3. Esto se debe a su alimentación: hay que considerar toda el agua necesaria para la producción de los pastos, cereales, etc, que el animal va a consumir durante toda su vida4. Es por ello que, para reducir la huella hídrica, es mejor consumir otro tipo de alimentos que nos permita alcanzar nuestros requerimientos nutricionales diarios.\")\n",
        "  st.write(\"Elije alimentos orgánicos\")\n",
        "  st.write(\"Muchos de los alimentos de cultivo que se consumen diariamente mantienen un proceso de producción que requieren riego constante. Algunos cereales, por ejemplo, emplean más de 1500 litros de agua en un solo kilogramo5. Por otra parte, los productos procesados también generan un gran impacto, ya que pasan por una cadena de limpieza y precocción. En ese sentido, el consumir alimentos orgánicos nos ayudará a reducir la huella hídrica debido a que los suelos en los que son sembrados permiten que se filtre y retengan eficientemente más agua, y no necesitan de ningún proceso ambientalmente perjudicial para su empaquetado6.\")\n",
        "  st.write(\"Revisa el etiquetado de la ropa\")\n",
        "  st.write(\"Actualmente, la industria textil es otro sector que consume grandes cantidades de agua para la fabricación de ropa. Informes hechos por El Corte Inglés, Fundación Botín y el Instituto Aitex indican que la ropa de los jóvenes es la que presenta mayor huella hídrica, pues su manufacturación requiere 15 mil litros de agua: las zapatillas requieren 4400 litros; un pantalón jean, 3000 litros; una camisa, 1000 litros; y un polo, 1200 litros7. Este gasto se debe al uso de algodón como insumo principal para las prendas. Ante esto, existen marcas certificadas que están empezando a ofrecer ropa en base a materiales reciclados y aguas residuales tratadas.\")\n",
        "\n",
        "if st.sidebar.button(\"del suelo\"):\n",
        "  col1.write(\"-Almacenamiento incorrecto de productos y/o residuos en actividades industriales, Vertidos de residuos incontrolados, Deposición de contaminantes atmosféricos\")\n",
        "  col2.write(\"Daños a la salud, Peores cultivos, Cambio climático, Contaminación de agua y aire, Desaparición de especies, Desertificación\")\n",
        "  st.write(\"Transporte sostenible\")\n",
        "  st.write(\"Se pueden utilizar alternativas sostenibles a los medios de transporte más contaminantes (motos y coches), como son la bicicleta, el transporte público, los patines o caminar. No conducir vehículos contaminantes puede llegar a reducir la huella ecológica de una persona hasta en un 20%.\") \n",
        "  st.write(\"Evita comprar productos con aceite de palma.\")\n",
        "  st.write(\"¿Te encanta desayunar bollería, galletas o cereales? ¿No imaginas un aperitivo sin patatas fritas? Mira bien los ingredientes de estos alimentos y si llevan aceite de palma, ¡descártalos! La palma y la soja están causando la deforestación de grandes zonas en selvas vírgenes en Indonesia, Brasil o Camerún. Además, es un cultivo muy agresivo con el suelo.\")\n",
        "  st.write(\"Evita comprar platos precocinados.\")\n",
        "  st.write(\"Contienen conservantes, colorantes o antiapelmazantes para mejorar su conservación y una gran cantidad de azúcar, para el cual se emplea mucha agua al producirlo. Prueba a preparar tus propias comidas, sándwiches o zumos, reducirás el consumo de azúcares, embalajes y plásticos.\")\n",
        "\n",
        "if st.sidebar.button(\"ambiental\"):\n",
        "  col1.write(\"Contaminantes químicos, Agentes físicos, Contaminantes biológicos\")\n",
        "  col2.write(\"Diversos tipos de contaminación, Daños en el estado de salud de los seres humanos, Desaparición de la capa de ozono, Daños en los ecosistemas\")\n",
        "  st.write(\"el crecimiento de la población, en la actualidad por el gran crecimiento de las familias, se han visto un vgran consumo en recursos y estos se ah estado sobre explotando\")\n",
        "  st.write(\"la pérdida de suelo fértil ya que estos se han estado contaminando o se quieran para construir, tomando en cuenta que solo unas partes son aptas para la agricultura\")\n",
        "  st.write(\"la deforestación esta tambien va relacionada con la sobre explotacion porque talamos mas arboles de los que se plantan y se dejan crecer\")\n",
        "  st.write(\"el aumento del consumo, esto va de la mano por el aumento de la poblacion.\")\n",
        "\n",
        "\n",
        "st.sidebar.write(\"participantes :\")\n",
        "st.sidebar.write(\"Pela Abigail Rodriguez Muñoz\")\n",
        "st.sidebar.write(\"Curso de Programnación\")\n",
        "st.sidebar.write(\"Facultad de Ciencias Quimicas, Univerdidad Autonoma de chihuahua\")\n",
        "st.sidebar.write(\"Docente: Jose Manuel Napoles Duarte\")\n",
        "\n",
        "\n",
        "\n",
        "st.title(\"Calculadora de huella ecologica\")\n",
        "number=st.slider(\"personas con las que vives\",1,10)\n",
        "\n",
        "size=st.select_slider(\"Que tan seguido consumes productos de origen animal\", options=[\"nunca\",\"casi nunca\", \"ocasionalmente\", \"diariamente\"],value=(\"nunca\"))\n",
        " \n",
        "st.write(\"nunca=0.1\")\n",
        "st.write(\"casi nunca=0.2\")\n",
        "st.write(\"ocasionalmente=1\")\n",
        "st.write(\"diariamente=2\")\n",
        "\n",
        "choice=st.selectbox(\"Porcentaje de tu comida no esta empaquetada\",[\"100\",\"75\",\"50\",\"25\",\"0\"])\n",
        "\n",
        "st.write(\"100=0.09\")\n",
        "st.write(\"75=0.15\")\n",
        "st.write(\"50=0.25\")\n",
        "st.write(\"25=0.45\")\n",
        "st.write(\"0=0.75\")\n",
        "\n",
        "choice=st.selectbox(\"Como es tu casa\",[\"independiente sin agua\",\"independiente con agua\",\"departamento con varios pisos\",\"duplex,edificio\",\"condominio de lujo\"])\n",
        "\n",
        "st.write(\"sin agua=0.2\")\n",
        "st.write(\"con agua=0.5\")\n",
        "st.write(\"varios pisos=0.3\")\n",
        "st.write(\"duplex=0.3\")\n",
        "st.write(\"condominio=0.8\")\n",
        "\n",
        "choice=st.selectbox(\"De que material esta construida\",[\"Adobe\",\"Madera\",\"Paja\",\"Ladrillo\",\"Acero/otro\"])\n",
        "choice=st.selectbox(\"tamaño de la vivienda\",[\"menor de 50\",\"50-140\",\"140-460\",\"mayor de 460\"])\n",
        "size=st.select_slider(\"Que tan eficiente es el consumo de energia de tu hogar\", options=[\"ineficiente\",\"promedio\", \"por encima del promedio\", \"muy eficiente\", \"no tengo electricidad\"],value=(\"no tengo electricidad\"))\n",
        "size=st.select_slider(\"Comparado con tus vecinos cuanta basura genera\", options=[\"mucho menos\",\"menos\", \"igual\", \"mas\", \"mucha mas\"],value=(\"mucho menos\"))\n",
        "choice=st.number_input(\"Que distancia viaja en transporte publico cada semana (km)\",0,100) \n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "background_save": true,
          "base_uri": "https://localhost:8080/"
        },
        "id": "aS-Fo8nYedKL",
        "outputId": "5bac013c-ce6c-4d2c-e7f8-db483308a57e"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "[##................] - fetchMetadata: sill resolveWithNewModule ms@2.1.2 checki\u001b[0m\u001b[K\n",
            "Collecting usage statistics. To deactivate, set browser.gatherUsageStats to False.\n",
            "\u001b[0m\n",
            "\u001b[0m\n",
            "\u001b[34m\u001b[1m  You can now view your Streamlit app in your browser.\u001b[0m\n",
            "\u001b[0m\n",
            "\u001b[34m  Network URL: \u001b[0m\u001b[1mhttp://172.28.0.2:8501\u001b[0m\n",
            "\u001b[34m  External URL: \u001b[0m\u001b[1mhttp://35.187.149.185:8501\u001b[0m\n",
            "\u001b[0m\n",
            "\u001b[K\u001b[?25hnpx: installed 22 in 7.078s\n",
            "your url is: https://evil-seas-design-35-187-149-185.loca.lt\n",
            "2022-11-10 22:31:31.541 Uncaught app exception\n",
            "Traceback (most recent call last):\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/streamlit/runtime/scriptrunner/script_runner.py\", line 563, in _run_script\n",
            "    exec(code, module.__dict__)\n",
            "  File \"/content/app.py\", line 17, in <module>\n",
            "    col1.writr(\"-Vertidos de aguas negras, Derrames de petróleo, Productos fitosanitarios, Deforestación, Aumento de temperatura\")\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/streamlit/delta_generator.py\", line 357, in wrapper\n",
            "    raise StreamlitAPIException(message)\n",
            "streamlit.errors.StreamlitAPIException: `writr()` is not a valid Streamlit command.\n",
            "2022-11-10 22:36:19.927 Uncaught app exception\n",
            "Traceback (most recent call last):\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/streamlit/runtime/scriptrunner/script_runner.py\", line 563, in _run_script\n",
            "    exec(code, module.__dict__)\n",
            "  File \"/content/app.py\", line 21, in <module>\n",
            "    col1.writr(\"Vertidos de aguas negras, Derrames de petróleo, Productos fitosanitarios, Deforestación, Aumento de temperatura\")\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/streamlit/delta_generator.py\", line 357, in wrapper\n",
            "    raise StreamlitAPIException(message)\n",
            "streamlit.errors.StreamlitAPIException: `writr()` is not a valid Streamlit command.\n",
            "2022-11-10 22:36:41.422 Uncaught app exception\n",
            "Traceback (most recent call last):\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/streamlit/runtime/scriptrunner/script_runner.py\", line 563, in _run_script\n",
            "    exec(code, module.__dict__)\n",
            "  File \"/content/app.py\", line 21, in <module>\n",
            "    col1.writr(\"Vertidos de aguas negras, Derrames de petróleo, Productos fitosanitarios, Deforestación, Aumento de temperatura\")\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/streamlit/delta_generator.py\", line 357, in wrapper\n",
            "    raise StreamlitAPIException(message)\n",
            "streamlit.errors.StreamlitAPIException: `writr()` is not a valid Streamlit command.\n",
            "2022-11-10 22:41:19.613 Uncaught app exception\n",
            "Traceback (most recent call last):\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/streamlit/runtime/scriptrunner/script_runner.py\", line 563, in _run_script\n",
            "    exec(code, module.__dict__)\n",
            "  File \"/content/app.py\", line 11, in <module>\n",
            "    col1, col2, col3 = st.columns([1,2])\n",
            "ValueError: not enough values to unpack (expected 3, got 2)\n",
            "2022-11-10 22:41:39.986 Uncaught app exception\n",
            "Traceback (most recent call last):\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/streamlit/runtime/scriptrunner/script_runner.py\", line 563, in _run_script\n",
            "    exec(code, module.__dict__)\n",
            "  File \"/content/app.py\", line 11, in <module>\n",
            "    col1, col2, col3 = st.columns([1,2])\n",
            "ValueError: not enough values to unpack (expected 3, got 2)\n",
            "2022-11-10 22:42:01.331 Uncaught app exception\n",
            "Traceback (most recent call last):\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/streamlit/runtime/scriptrunner/script_runner.py\", line 563, in _run_script\n",
            "    exec(code, module.__dict__)\n",
            "  File \"/content/app.py\", line 11, in <module>\n",
            "    col1, col2, col3 = st.columns([1,2])\n",
            "ValueError: not enough values to unpack (expected 3, got 2)\n",
            "2022-11-10 22:43:19.655 Uncaught app exception\n",
            "Traceback (most recent call last):\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/streamlit/runtime/scriptrunner/script_runner.py\", line 563, in _run_script\n",
            "    exec(code, module.__dict__)\n",
            "  File \"/content/app.py\", line 16, in <module>\n",
            "    st.header()\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/streamlit/runtime/metrics_util.py\", line 311, in wrapped_func\n",
            "    result = non_optional_func(*args, **kwargs)\n",
            "TypeError: header() missing 1 required positional argument: 'body'\n",
            "2022-11-10 22:48:49.645 Uncaught app exception\n",
            "Traceback (most recent call last):\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/streamlit/runtime/scriptrunner/script_runner.py\", line 563, in _run_script\n",
            "    exec(code, module.__dict__)\n",
            "  File \"/content/app.py\", line 23, in <module>\n",
            "    if st.sidebar.button(\"Hidrica\"):\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/streamlit/runtime/metrics_util.py\", line 311, in wrapped_func\n",
            "    result = non_optional_func(*args, **kwargs)\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/streamlit/elements/button.py\", line 140, in button\n",
            "    ctx=ctx,\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/streamlit/elements/button.py\", line 375, in _button\n",
            "    ctx=ctx,\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/streamlit/runtime/state/widgets.py\", line 212, in register_widget\n",
            "    user_key,\n",
            "streamlit.errors.DuplicateWidgetID: There are multiple identical `st.button` widgets with the\n",
            "same generated key.\n",
            "\n",
            "When a widget is created, it's assigned an internal key based on\n",
            "its structure. Multiple widgets with an identical structure will\n",
            "result in the same internal key, which causes this error.\n",
            "\n",
            "To fix this error, please pass a unique `key` argument to\n",
            "`st.button`.\n",
            "2022-11-10 22:48:56.967 Uncaught app exception\n",
            "Traceback (most recent call last):\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/streamlit/runtime/scriptrunner/script_runner.py\", line 563, in _run_script\n",
            "    exec(code, module.__dict__)\n",
            "  File \"/content/app.py\", line 28, in <module>\n",
            "    if st.sidebar.button(\"Hidrica\"):\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/streamlit/runtime/metrics_util.py\", line 311, in wrapped_func\n",
            "    result = non_optional_func(*args, **kwargs)\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/streamlit/elements/button.py\", line 140, in button\n",
            "    ctx=ctx,\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/streamlit/elements/button.py\", line 375, in _button\n",
            "    ctx=ctx,\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/streamlit/runtime/state/widgets.py\", line 212, in register_widget\n",
            "    user_key,\n",
            "streamlit.errors.DuplicateWidgetID: There are multiple identical `st.button` widgets with the\n",
            "same generated key.\n",
            "\n",
            "When a widget is created, it's assigned an internal key based on\n",
            "its structure. Multiple widgets with an identical structure will\n",
            "result in the same internal key, which causes this error.\n",
            "\n",
            "To fix this error, please pass a unique `key` argument to\n",
            "`st.button`.\n",
            "2022-11-10 22:50:02.566 Uncaught app exception\n",
            "Traceback (most recent call last):\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/streamlit/runtime/scriptrunner/script_runner.py\", line 563, in _run_script\n",
            "    exec(code, module.__dict__)\n",
            "  File \"/content/app.py\", line 28, in <module>\n",
            "    if st.sidebar.button(\"Hidrica\"):\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/streamlit/runtime/metrics_util.py\", line 311, in wrapped_func\n",
            "    result = non_optional_func(*args, **kwargs)\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/streamlit/elements/button.py\", line 140, in button\n",
            "    ctx=ctx,\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/streamlit/elements/button.py\", line 375, in _button\n",
            "    ctx=ctx,\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/streamlit/runtime/state/widgets.py\", line 212, in register_widget\n",
            "    user_key,\n",
            "streamlit.errors.DuplicateWidgetID: There are multiple identical `st.button` widgets with the\n",
            "same generated key.\n",
            "\n",
            "When a widget is created, it's assigned an internal key based on\n",
            "its structure. Multiple widgets with an identical structure will\n",
            "result in the same internal key, which causes this error.\n",
            "\n",
            "To fix this error, please pass a unique `key` argument to\n",
            "`st.button`.\n",
            "2022-11-10 22:55:11.312 Uncaught app exception\n",
            "Traceback (most recent call last):\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/streamlit/runtime/scriptrunner/script_runner.py\", line 563, in _run_script\n",
            "    exec(code, module.__dict__)\n",
            "  File \"/content/app.py\", line 4, in <module>\n",
            "    st.subtitle(\"*$¿Qué$ $es$ $el$ $calentamiento$ $global?$*\")\n",
            "AttributeError: module 'streamlit' has no attribute 'subtitle'\n",
            "2022-11-10 23:12:49.728 Uncaught app exception\n",
            "Traceback (most recent call last):\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/streamlit/runtime/scriptrunner/script_runner.py\", line 563, in _run_script\n",
            "    exec(code, module.__dict__)\n",
            "  File \"/content/app.py\", line 8, in <module>\n",
            "    st.ca(\"-huella de carbono\")\n",
            "AttributeError: module 'streamlit' has no attribute 'ca'\n",
            "2022-11-10 23:17:58.872 Uncaught app exception\n",
            "Traceback (most recent call last):\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/streamlit/runtime/scriptrunner/script_runner.py\", line 563, in _run_script\n",
            "    exec(code, module.__dict__)\n",
            "  File \"/content/app.py\", line 36, in <module>\n",
            "    st,write(\"Actualmente, la industria textil es otro sector que consume grandes cantidades de agua para la fabricación de ropa. Informes hechos por El Corte Inglés, Fundación Botín y el Instituto Aitex indican que la ropa de los jóvenes es la que presenta mayor huella hídrica, pues su manufacturación requiere 15 mil litros de agua: las zapatillas requieren 4400 litros; un pantalón jean, 3000 litros; una camisa, 1000 litros; y un polo, 1200 litros7. Este gasto se debe al uso de algodón como insumo principal para las prendas. Ante esto, existen marcas certificadas que están empezando a ofrecer ropa en base a materiales reciclados y aguas residuales tratadas.\")\n",
            "NameError: name 'write' is not defined\n",
            "2022-11-10 23:51:35.762 Uncaught app exception\n",
            "Traceback (most recent call last):\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/streamlit/runtime/scriptrunner/script_runner.py\", line 563, in _run_script\n",
            "    exec(code, module.__dict__)\n",
            "  File \"/content/app.py\", line 53, in <module>\n",
            "    st.writw(\"la deforestación\")\n",
            "AttributeError: module 'streamlit' has no attribute 'writw'\n",
            "2022-11-11 00:52:21.862 Uncaught app exception\n",
            "Traceback (most recent call last):\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/streamlit/runtime/scriptrunner/script_runner.py\", line 563, in _run_script\n",
            "    exec(code, module.__dict__)\n",
            "  File \"/content/app.py\", line 70, in <module>\n",
            "    number=st.slider(\"consumo de energia(mensual)\",kWn)\n",
            "NameError: name 'kWn' is not defined\n",
            "2022-11-11 00:57:31.138 `label` got an empty value. This is discouraged for accessibility reasons and may be disallowed in the future by raising an exception. Please provide a non-empty label and hide it with label_visibility if needed.\n",
            "2022-11-11 00:58:31.142 Uncaught app exception\n",
            "Traceback (most recent call last):\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/streamlit/runtime/scriptrunner/script_runner.py\", line 563, in _run_script\n",
            "    exec(code, module.__dict__)\n",
            "  File \"/content/app.py\", line 71, in <module>\n",
            "    size=st.select_slid(\"alimentos de origen animal\",)\n",
            "AttributeError: module 'streamlit' has no attribute 'select_slid'\n",
            "2022-11-11 01:57:14.921 Uncaught app exception\n",
            "Traceback (most recent call last):\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/streamlit/runtime/scriptrunner/script_runner.py\", line 563, in _run_script\n",
            "    exec(code, module.__dict__)\n",
            "  File \"/content/app.py\", line 71, in <module>\n",
            "    size=st.select_slider(\"alimentos de origen animal\", options=[\"nada\",\"poco\",  \"mucho\", \"demasiado\", \"excesivo\"],value=(\"poco\",\"e\"))\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/streamlit/runtime/metrics_util.py\", line 311, in wrapped_func\n",
            "    result = non_optional_func(*args, **kwargs)\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/streamlit/elements/select_slider.py\", line 215, in select_slider\n",
            "    ctx=ctx,\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/streamlit/elements/select_slider.py\", line 260, in _select_slider\n",
            "    slider_value = as_index_list(value)\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/streamlit/elements/select_slider.py\", line 244, in as_index_list\n",
            "    slider_value = [index_(opt, val) for val in v]\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/streamlit/elements/select_slider.py\", line 244, in <listcomp>\n",
            "    slider_value = [index_(opt, val) for val in v]\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/streamlit/util.py\", line 140, in index_\n",
            "    raise ValueError(\"{} is not in iterable\".format(str(x)))\n",
            "ValueError: e is not in iterable\n",
            "2022-11-11 01:59:40.517 Uncaught app exception\n",
            "Traceback (most recent call last):\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/streamlit/runtime/scriptrunner/script_runner.py\", line 563, in _run_script\n",
            "    exec(code, module.__dict__)\n",
            "  File \"/content/app.py\", line 72, in <module>\n",
            "    choice=st.number_imput\n",
            "AttributeError: module 'streamlit' has no attribute 'number_imput'\n",
            "2022-11-11 01:59:44.929 Uncaught app exception\n",
            "Traceback (most recent call last):\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/streamlit/runtime/scriptrunner/script_runner.py\", line 563, in _run_script\n",
            "    exec(code, module.__dict__)\n",
            "  File \"/content/app.py\", line 72, in <module>\n",
            "    choice=st.number_imput()\n",
            "AttributeError: module 'streamlit' has no attribute 'number_imput'\n",
            "2022-11-11 02:00:14.910 Uncaught app exception\n",
            "Traceback (most recent call last):\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/streamlit/runtime/scriptrunner/script_runner.py\", line 563, in _run_script\n",
            "    exec(code, module.__dict__)\n",
            "  File \"/content/app.py\", line 72, in <module>\n",
            "    choice=st.number_imput(\"metros cuadrados de la vivienda\",140,)\n",
            "AttributeError: module 'streamlit' has no attribute 'number_imput'\n",
            "2022-11-11 02:03:14.902 Uncaught app exception\n",
            "Traceback (most recent call last):\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/streamlit/runtime/scriptrunner/script_runner.py\", line 563, in _run_script\n",
            "    exec(code, module.__dict__)\n",
            "  File \"/content/app.py\", line 72, in <module>\n",
            "    choice=st.number_imput(\"metros cuadrados de la vivienda\",140,250)\n",
            "AttributeError: module 'streamlit' has no attribute 'number_imput'\n",
            "2022-11-11 02:03:18.583 Uncaught app exception\n",
            "Traceback (most recent call last):\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/streamlit/runtime/scriptrunner/script_runner.py\", line 563, in _run_script\n",
            "    exec(code, module.__dict__)\n",
            "  File \"/content/app.py\", line 72, in <module>\n",
            "    choice=st.number_imput(\"metros cuadrados de la vivienda\",140,250)\n",
            "AttributeError: module 'streamlit' has no attribute 'number_imput'\n",
            "2022-11-11 02:19:45.036 Uncaught app exception\n",
            "Traceback (most recent call last):\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/streamlit/runtime/scriptrunner/script_runner.py\", line 563, in _run_script\n",
            "    exec(code, module.__dict__)\n",
            "  File \"/content/app.py\", line 75, in <module>\n",
            "    size=st.select_slider(\"alimentos de origen animal\", options=[\"nada\",\"poco\",  \"mucho\", \"demasiado\", \"excesivo\"],value=(\"nada\"))\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/streamlit/runtime/metrics_util.py\", line 311, in wrapped_func\n",
            "    result = non_optional_func(*args, **kwargs)\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/streamlit/elements/select_slider.py\", line 215, in select_slider\n",
            "    ctx=ctx,\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/streamlit/elements/select_slider.py\", line 286, in _select_slider\n",
            "    ctx=ctx,\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/streamlit/runtime/state/widgets.py\", line 212, in register_widget\n",
            "    user_key,\n",
            "streamlit.errors.DuplicateWidgetID: There are multiple identical `st.slider` widgets with the\n",
            "same generated key.\n",
            "\n",
            "When a widget is created, it's assigned an internal key based on\n",
            "its structure. Multiple widgets with an identical structure will\n",
            "result in the same internal key, which causes this error.\n",
            "\n",
            "To fix this error, please pass a unique `key` argument to\n",
            "`st.slider`.\n",
            "2022-11-11 02:20:14.909 Uncaught app exception\n",
            "Traceback (most recent call last):\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/streamlit/runtime/scriptrunner/script_runner.py\", line 563, in _run_script\n",
            "    exec(code, module.__dict__)\n",
            "  File \"/content/app.py\", line 73, in <module>\n",
            "    size=st.select_slider(\"Que tan seguido consumes productos de origen animal\", options=[\"nunca\",\"poco\",  \"mucho\", \"demasiado\", \"excesivo\"],value=(\"nada\"))\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/streamlit/runtime/metrics_util.py\", line 311, in wrapped_func\n",
            "    result = non_optional_func(*args, **kwargs)\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/streamlit/elements/select_slider.py\", line 215, in select_slider\n",
            "    ctx=ctx,\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/streamlit/elements/select_slider.py\", line 260, in _select_slider\n",
            "    slider_value = as_index_list(value)\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/streamlit/elements/select_slider.py\", line 252, in as_index_list\n",
            "    return [index_(opt, v)]\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/streamlit/util.py\", line 140, in index_\n",
            "    raise ValueError(\"{} is not in iterable\".format(str(x)))\n",
            "ValueError: nada is not in iterable\n",
            "2022-11-11 02:20:45.037 Uncaught app exception\n",
            "Traceback (most recent call last):\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/streamlit/runtime/scriptrunner/script_runner.py\", line 563, in _run_script\n",
            "    exec(code, module.__dict__)\n",
            "  File \"/content/app.py\", line 73, in <module>\n",
            "    size=st.select_slider(\"Que tan seguido consumes productos de origen animal\", options=[\"nunca\",\"casi nunca\", \"ocasionalmente\", \"diariamente\"],value=(\"nada\"))\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/streamlit/runtime/metrics_util.py\", line 311, in wrapped_func\n",
            "    result = non_optional_func(*args, **kwargs)\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/streamlit/elements/select_slider.py\", line 215, in select_slider\n",
            "    ctx=ctx,\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/streamlit/elements/select_slider.py\", line 260, in _select_slider\n",
            "    slider_value = as_index_list(value)\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/streamlit/elements/select_slider.py\", line 252, in as_index_list\n",
            "    return [index_(opt, v)]\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/streamlit/util.py\", line 140, in index_\n",
            "    raise ValueError(\"{} is not in iterable\".format(str(x)))\n",
            "ValueError: nada is not in iterable\n",
            "2022-11-11 02:21:14.923 Uncaught app exception\n",
            "Traceback (most recent call last):\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/streamlit/runtime/scriptrunner/script_runner.py\", line 563, in _run_script\n",
            "    exec(code, module.__dict__)\n",
            "  File \"/content/app.py\", line 74, in <module>\n",
            "    num\n",
            "NameError: name 'num' is not defined\n",
            "2022-11-11 02:22:44.923 Uncaught app exception\n",
            "Traceback (most recent call last):\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/streamlit/runtime/scriptrunner/script_runner.py\", line 563, in _run_script\n",
            "    exec(code, module.__dict__)\n",
            "  File \"/content/app.py\", line 74, in <module>\n",
            "    choice=st.radio(\"Porcentaje de tu comida no esta emaquetad\")\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/streamlit/runtime/metrics_util.py\", line 311, in wrapped_func\n",
            "    result = non_optional_func(*args, **kwargs)\n",
            "TypeError: radio() missing 1 required positional argument: 'options'\n",
            "2022-11-11 02:23:14.919 Uncaught app exception\n",
            "Traceback (most recent call last):\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/streamlit/runtime/scriptrunner/script_runner.py\", line 563, in _run_script\n",
            "    exec(code, module.__dict__)\n",
            "  File \"/content/app.py\", line 74, in <module>\n",
            "    choice=st.radio(\"Porcentaje de tu comida no esta empaquetada\")\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/streamlit/runtime/metrics_util.py\", line 311, in wrapped_func\n",
            "    result = non_optional_func(*args, **kwargs)\n",
            "TypeError: radio() missing 1 required positional argument: 'options'\n",
            "2022-11-11 02:24:33.156 Uncaught app exception\n",
            "Traceback (most recent call last):\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/streamlit/runtime/scriptrunner/script_runner.py\", line 563, in _run_script\n",
            "    exec(code, module.__dict__)\n",
            "  File \"/content/app.py\", line 74, in <module>\n",
            "    choice=st.radio(\"Porcentaje de tu comida no esta empaquetada\",[\"100\",\"75\",\"50\",\"25\",\"0\"],value=(\"0\"))\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/streamlit/runtime/metrics_util.py\", line 311, in wrapped_func\n",
            "    result = non_optional_func(*args, **kwargs)\n",
            "TypeError: radio() got an unexpected keyword argument 'value'\n",
            "2022-11-11 02:29:44.936 Uncaught app exception\n",
            "Traceback (most recent call last):\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/streamlit/runtime/scriptrunner/script_runner.py\", line 563, in _run_script\n",
            "    exec(code, module.__dict__)\n",
            "  File \"/content/app.py\", line 76, in <module>\n",
            "    choice=st.selectbox(\"Porcentaje de tu comida no esta empaquetada\",[\"100\",\"75\",\"50\",\"25\",\"0\"])\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/streamlit/runtime/metrics_util.py\", line 311, in wrapped_func\n",
            "    result = non_optional_func(*args, **kwargs)\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/streamlit/elements/selectbox.py\", line 157, in selectbox\n",
            "    ctx=ctx,\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/streamlit/elements/selectbox.py\", line 213, in _selectbox\n",
            "    ctx=ctx,\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/streamlit/runtime/state/widgets.py\", line 212, in register_widget\n",
            "    user_key,\n",
            "streamlit.errors.DuplicateWidgetID: There are multiple identical `st.selectbox` widgets with the\n",
            "same generated key.\n",
            "\n",
            "When a widget is created, it's assigned an internal key based on\n",
            "its structure. Multiple widgets with an identical structure will\n",
            "result in the same internal key, which causes this error.\n",
            "\n",
            "To fix this error, please pass a unique `key` argument to\n",
            "`st.selectbox`.\n",
            "2022-11-11 02:34:44.958 Uncaught app exception\n",
            "Traceback (most recent call last):\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/streamlit/runtime/scriptrunner/script_runner.py\", line 563, in _run_script\n",
            "    exec(code, module.__dict__)\n",
            "  File \"/content/app.py\", line 77, in <module>\n",
            "    choice=st.selectbox(\"Porcentaje de tu comida no esta empaquetada\",[\"100\",\"75\",\"50\",\"25\",\"0\"])\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/streamlit/runtime/metrics_util.py\", line 311, in wrapped_func\n",
            "    result = non_optional_func(*args, **kwargs)\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/streamlit/elements/selectbox.py\", line 157, in selectbox\n",
            "    ctx=ctx,\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/streamlit/elements/selectbox.py\", line 213, in _selectbox\n",
            "    ctx=ctx,\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/streamlit/runtime/state/widgets.py\", line 212, in register_widget\n",
            "    user_key,\n",
            "streamlit.errors.DuplicateWidgetID: There are multiple identical `st.selectbox` widgets with the\n",
            "same generated key.\n",
            "\n",
            "When a widget is created, it's assigned an internal key based on\n",
            "its structure. Multiple widgets with an identical structure will\n",
            "result in the same internal key, which causes this error.\n",
            "\n",
            "To fix this error, please pass a unique `key` argument to\n",
            "`st.selectbox`.\n",
            "2022-11-11 02:37:14.949 Uncaught app exception\n",
            "Traceback (most recent call last):\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/streamlit/runtime/scriptrunner/script_runner.py\", line 563, in _run_script\n",
            "    exec(code, module.__dict__)\n",
            "  File \"/content/app.py\", line 78, in <module>\n",
            "    size=st.select_slider(\"Que tan eficiente es el consumo de energia de tu hogar\", options=[\"ineficiente\",\"promedio\", \"por encima del promedio\", \"muy eficiente\", \"excesivo\"],value=(\"nada\"))\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/streamlit/runtime/metrics_util.py\", line 311, in wrapped_func\n",
            "    result = non_optional_func(*args, **kwargs)\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/streamlit/elements/select_slider.py\", line 215, in select_slider\n",
            "    ctx=ctx,\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/streamlit/elements/select_slider.py\", line 260, in _select_slider\n",
            "    slider_value = as_index_list(value)\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/streamlit/elements/select_slider.py\", line 252, in as_index_list\n",
            "    return [index_(opt, v)]\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/streamlit/util.py\", line 140, in index_\n",
            "    raise ValueError(\"{} is not in iterable\".format(str(x)))\n",
            "ValueError: nada is not in iterable\n",
            "2022-11-11 02:39:14.940 Uncaught app exception\n",
            "Traceback (most recent call last):\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/streamlit/runtime/scriptrunner/script_runner.py\", line 563, in _run_script\n",
            "    exec(code, module.__dict__)\n",
            "  File \"/content/app.py\", line 79, in <module>\n",
            "    size=st.select_slider(\"Comparado con tus vecinos cuanta basura genera\", options=[\"mucho menos\",\"menos\", \"igual\", \"mas\", \"much\"],value=(\"no tengo electricidad\"))\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/streamlit/runtime/metrics_util.py\", line 311, in wrapped_func\n",
            "    result = non_optional_func(*args, **kwargs)\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/streamlit/elements/select_slider.py\", line 215, in select_slider\n",
            "    ctx=ctx,\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/streamlit/elements/select_slider.py\", line 260, in _select_slider\n",
            "    slider_value = as_index_list(value)\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/streamlit/elements/select_slider.py\", line 252, in as_index_list\n",
            "    return [index_(opt, v)]\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/streamlit/util.py\", line 140, in index_\n",
            "    raise ValueError(\"{} is not in iterable\".format(str(x)))\n",
            "ValueError: no tengo electricidad is not in iterable\n",
            "2022-11-11 02:49:44.946 Uncaught app exception\n",
            "Traceback (most recent call last):\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/streamlit/runtime/scriptrunner/script_runner.py\", line 563, in _run_script\n",
            "    exec(code, module.__dict__)\n",
            "  File \"/content/app.py\", line 75, in <module>\n",
            "    porcentaje\n",
            "NameError: name 'porcentaje' is not defined\n",
            "2022-11-11 02:50:14.952 Uncaught app exception\n",
            "Traceback (most recent call last):\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/streamlit/runtime/scriptrunner/script_runner.py\", line 563, in _run_script\n",
            "    exec(code, module.__dict__)\n",
            "  File \"/content/app.py\", line 75, in <module>\n",
            "    nunca\n",
            "NameError: name 'nunca' is not defined\n",
            "2022-11-11 03:01:30.673 Uncaught app exception\n",
            "Traceback (most recent call last):\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/streamlit/runtime/scriptrunner/script_runner.py\", line 563, in _run_script\n",
            "    exec(code, module.__dict__)\n",
            "  File \"/content/app.py\", line 83, in <module>\n",
            "    st,write(\"100=0.09\")\n",
            "NameError: name 'write' is not defined\n",
            "2022-11-11 03:02:08.669 Uncaught app exception\n",
            "Traceback (most recent call last):\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/streamlit/runtime/scriptrunner/script_runner.py\", line 563, in _run_script\n",
            "    exec(code, module.__dict__)\n",
            "  File \"/content/app.py\", line 83, in <module>\n",
            "    st,write(\"100=0.09\")\n",
            "NameError: name 'write' is not defined\n"
          ]
        }
      ],
      "source": [
        "!streamlit run app.py & npx localtunnel --port 8501"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "SdrHktcSOxZn"
      },
      "outputs": [],
      "source": []
    }
  ],
  "metadata": {
    "colab": {
      "collapsed_sections": [],
      "provenance": []
    },
    "kernelspec": {
      "display_name": "Python 3",
      "name": "python3"
    },
    "language_info": {
      "name": "python"
    }
  },
  "nbformat": 4,
  "nbformat_minor": 0
}